typing/typecore.ml

```diff
@@ -6879,18 +6878,20 @@ and type_generic_array
       ~expected_mode
       ~ty_expected
       ~explanation
-      ~type_
       ~mutability
       ~attributes
       sargl
   =
+  let type_, base_argument_mode = match mutability with
+    | Mutable -> Predef.type_array, mode_default Value_mode.global
+    | Immutable -> Predef.type_iarray, mode_subcomponent expected_mode
```

Comment:
@riaqn: Does the `mode_subcomponent` here look right to you? Feel free to message me or Antal to explain more context here.

Reply:
I'm terribly sorry about the delay, yes it looks correct to me. A value being `regional` means it's anywhere except the current region. If an array is `regional`, we require its content to be `global`, because requiring `regional` is too vague - the content might be in a region "inner" than the array and get deleted, in which case the array would be pointing to an invalid address. @lpw25 will correct me if I'm wrong.